app/Repositories/DescriptionRepository.php

```diff
@@ -3,13 +3,14 @@
 namespace App\Repositories;
 
 use App\Description;
+use App\Http\Requests\DescriptionRequest;
 
 class DescriptionRepository
 {
-    public function make(string $text, $user): Description
+    public function make(DescriptionRequest $request, $user): Description
```

Comment:
Som jeg skrev, er jeg ikke så glad for Request eller andre "container" i repositories. Det er bedre hvis her så indeholder funktionsargumenteren `make(string text, $user)` - så det er mere "frit" i dens anvendelse, uden at jeg behøver at wrapper en text i DescriptionRequest før jeg kalder `DescriptionRepository->make()` - Det giver perfekt mening at bruge Request i services - men ikke rigtig i Repositories... IMO

Reply:
Hvordan kan du validere text, hvis du har tænkt dig at oprette en description uden om request?

Reply:
Det er allerede valideret i service....

Reply:
Hvis du mener validateDescription i servicen, er den erstattes til DescriptionRequest

Reply:
Ja - og da det er indelagt DescriptionRequest der i metoden (i service), kan du så kalde `$request->validated();`, og alt indholdet er så validateret i henhold til den rule du har sat op. Så her kan du glemme om validationslogik, og blot bruge $text og $user til at oprette en description.

Reply:
Jeg kan stadig ikke se hvorfor du vil have at parametre skal bestå af primitive variabler. I Endpoint bliver der ALTID initialisere en ny request som påfyldes med alle påkrævede variabler som bringes videre til controlleren, derefter repository direkte eller via service. Så requesten er aldrig inden for scopen, så jeg forstår ikke at du har behov for at signaturen i make() skal bestå af primitive. Jeg kan hellere ikke se andre steder, hvor vi kommer til at ville oprette en description som ikke kommer fra endpoint.

Reply:
Lad mig komme med en par eksempler:
- Vi vil gerne muliggør i Nova (eller tilføje et eller andet administrationsinterface til redaktørerne) så administratorerne kan redigere i Description. Denne gang vælger vi f.eks. ved dovenhed skyld at vælge at implementere det i plain HTML. Inputs vi får fra form er ikke ren description form request, så vi må trække teksten fra den request, initalisere en ny DescriptionRequest for at så endelig at kalde `make()`. Der kræver så 2 forskellige requests.
- Vi vil gerne oprette en command (`php artisan xxx`). Det skal kunne indsætte description til 1000 videoer vha. en simpel input, eller fra en csv fil eller whatever. Der har vi af naturens sag intet med requests at gøre, men af rent fil-input eller af manuelt indtastet desc og en array med video id's. Skal vi så sætte dem ind i hver sin DescriptionRequest objekt før de endelig kan anvendes i `make`?
Min pointe er at ovenstående skal ikke ende med at skulle oprette en 100% ens metode i repository, hvis de tjener den samme formål. Det eneste som `make` metoden i repositorien har brug for, er selve literalen, og selve brugeren. Slet ikke andet, så det behøves ikke at wrappes i et eller andet udenom. Jeg har forestillet mig at alle repositories er IKKE afhængigt af hvordan data er kommet ind, men "skære ind i benet" og kun kræver de allermeste nødvendige data at bruge til at oprette lige præcis den dataobjekt vi har brug for. Det følger så KISS-princippet.

Jeg kan godt se din pointe at det er ikke nødvendigt nu. Vi tænker i to forskellige måder. Du foretrækker at udvikle metoderne løbende som vi har brug for dem, mens jeg foretrækker at vi allerede fra starten af bygger en god og solid fundament så vi ikke behøver at gå tilbage og ændre i hele kodebasen fordi vi gerne vil tilføje en adfærd vi ikke forventede. ...